### PR TITLE
Microservices RA - Added ingress TLS secrets to keyvault - delivery service

### PR DESCRIPTION
**Pull request by magrande**

### CODE REVIEW

File: charts/delivery/templates/delivery-deploy.yaml [hunk 1]
```diff
@@ -85,7 +85,21 @@ spec:
...
+        - name: INGRESS_TLS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: delivery-secrets
+              key: ingress-tls-key
+        - name: INGRESS_TLS_CRT
+          valueFrom:
+            secretKeyRef:
+              name: delivery-secrets
+              key: ingress-tls-crt
```
Comment (ckittel) [inline charts/delivery/templates/delivery-deploy.yaml:102]: These used to be in a k8s `Secret`, how did the workload get access to these before?  Anything need to change about that process? They weren't being mounted before as env variables.  Is the workload trying to access them directly?  I'm a bit confused on where the "second part" of the change is here.

Comment (ckittel) [reply charts/delivery/templates/delivery-deploy.yaml:102]: I see that the secret name is referenced in the ingress controller as `tlsSecretName`, right?  Why do they need to be mounted as part of the delivery workload here?

**Review by ckittel — CHANGES_REQUESTED** [review]
Mostly just an implementation question/clarification for me.

File: deployment.md [hunk 2]
```diff
@@ -249,6 +251,10 @@ spec:
       key: queue-keyvalue
     - objectName: ApplicationInsights--InstrumentationKey
       key: appinsights-ikey
+    - objectName: Ingestion-Ingress-Tls-Key
+      key: ingestion-ingress-tls.key
```
Comment (magrande) [inline deployment.md:255]: @ckittel  I'm not sure this is the name of the key, Can you help me figuring out where it's being used ?

Comment (ckittel) [reply deployment.md:255]: Responded on call. The names you're looking for are in the secrets files that you deleted as part of this PR.

File: deployment.md [hunk 3]
```diff
@@ -515,9 +567,22 @@ Build the Ingestion service.
...
+export SIGNED_IN_OBJECT_ID=$(az ad signed-in-user show --query 'objectId' -o tsv)
+
+az keyvault set-policy --secret-permissions set --object-id $SIGNED_IN_OBJECT_ID -n $INGESTION_KEYVAULT_NAME
+az keyvault secret set --name Ingestion-Ingress-Tls-Key --vault-name $INGESTION_KEYVAULT_NAME --value "$(cat ingestion-ingress-tls.key)"
+az keyvault secret set --name Ingestion-Ingress-Tls-Crt --vault-name $INGESTION_KEYVAULT_NAME --value "$(cat ingestion-ingress-tls.crt)"
+az keyvault delete-policy --object-id $SIGNED_IN_OBJECT_ID -n $INGESTION_KEYVAULT_NAME
+
+
 # Set secreat name
 export INGRESS_TLS_SECRET_NAME=ingestion-ingress-tls
```
Comment (ckittel) [inline deployment.md:587]: How is this so low in the instructions?  You needed to use this value both in the helm chart right below, but also when you created the `SecretProviderClass`.  Looks like you are calling those secrets the wrong name though, `ingestion-secrets` -- hard coded.

**Review by ckittel — CHANGES_REQUESTED** [review]

File: deployment.md [hunk 4]
```diff
@@ -320,7 +331,7 @@ metadata:
 spec:
   provider: azure
   secretObjects:
-  - secretName: "${DELIVERY_INGRESS_TLS_SECRET_NAME}"
+  - secretName: "${INGRESS_TLS_SECRET_NAME}"
```
Comment (ckittel) [inline deployment.md:334]: Why did this change now?  We now have two `SecretProviderClass` instances managing the exact same secret.

**Review by ckittel — CHANGES_REQUESTED** [review]

File: deployment.md (hunk 3 above)
Comment (ckittel) [reply deployment.md]: Moved out, resolving.  Opened another item below for a side effect of your last change.

**Review by ckittel — APPROVED** [review]
Assuming you're able to complete your testing, this change looks good.  Marking approved, just let me know when testing is complete.

**Comment by magrande** [comment]
> Assuming you're able to complete your testing, this change looks good. Marking approved, just let me know when testing is complete.

@ckittel  I had to do some "subtle" change in the secretsprovider class, after that, Ingestion service started to work properly.

**Pull request merged**